pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotWindowExchangeNodeInsertRule.java

```diff
@@ -160,6 +162,19 @@ private boolean isPartitionByOnlyQuery(Window.Group windowGroup) {
       Set<Integer> partitionByKeyList = new HashSet<>(windowGroup.keys.toList());
       Set<Integer> orderByKeyList = new HashSet<>(windowGroup.orderKeys.getKeys());
       isPartitionByOnly = partitionByKeyList.equals(orderByKeyList);
+      if (isPartitionByOnly) {
```

Comment:
I'm not entirely sure if this is actually required, but thought I'd look into the correct semantics once I start the work on `ORDER BY` within `OVER()`. This check will prevent marking queries like below as partition by only queries:

- OVER(PARTITION BY key1 ORDER BY key1 DESC)
- OVER(PARTITION BY key1 ORDER BY key1 NULLS FIRST)

The code will however allow queries of the type `OVER(PARTITION BY key1 ORDER BY key1)` to be marked as partition by only (since all aggregated values will be the same across the whole partition just like OVER(PARTITION BY)).

Let me know if the preference is to just remove this extra check since logically the actual aggregation values will still be the same for the full partition irrespective of the direction and null direction.

Reply:
all 3 of the example above produces the exact same query results right?

Reply:
yup, i tried these in postgres, but since postgres does an explicit sort for even the PARTITION BY keys it did make a difference in the ordering.

Since we've decided not to care about ordering on PARTITION BY key, the above 3 queries will have the same aggregation value. The NULLS LAST vs. NULLS FIRST will only modify the order of the NULL rows, but this can be handled within the operator and still treat this type of query as a PARTITION BY only query.

I guess based on my own response above, adding this check probably doesn't make sense 😅

Reply:
@walterddr should I go ahead and remove this check?

Reply:
+1 to removing this because:
1. We use isPartitionByOnlyQuery to decide if we have to use LogicalExchange vs LogicalSortExchange
2. Based on my understanding, no sorting of rows within a partition is needed if the partition key and order by key are the same.

Reply:
done, removed this check. I'll add special handling for null direction later on (left a TODO). This will only affect the order in which the columns with NULL values are output in the final result.